datadog_checks_base/datadog_checks/base/utils/time.py

```diff
@@ -6,61 +6,41 @@
 from datetime import datetime
 from time import time as epoch_offset
 
-import pytz
-from six import PY2
+from dateutil.tz import UTC
 
-UTC = pytz.utc
-EPOCH = datetime(1970, 1, 1, tzinfo=UTC)
+EPOCH = datetime.fromtimestamp(0, UTC)
 
 
-if PY2:
-
-    def get_timestamp(dt=None):
-        """
-        Returns the number of seconds since the Unix epoch.
-        If `dt` is not specified or `None`, the current time in UTC is assumed.
-        """
-        if dt is None:
-            return epoch_offset()
-
-        return (dt - EPOCH).total_seconds()
-
-
-else:
-
-    def get_timestamp(dt=None):
-        """
-        Returns the number of seconds since the Unix epoch.
-        If `dt` is not specified or `None`, the current time in UTC is assumed.
-        """
-        if dt is None:
-            # NOTE: Although the epoch is platform dependent, it appears to be the same
-            # for all platforms we've tested, therefore we use `time.time` for speed.
-            #
-            # Here is the test:
-            #
-            #     $ python -c "import time;print(tuple(time.gmtime(0)[:9]))"
-            #     (1970, 1, 1, 0, 0, 0, 3, 1, 0)
-            #
-            # If you can reproduce, add to the following list of tested platforms:
-            #
-            # - Windows
-            # - macOS
-            # - Ubuntu
-            # - Alpine
-            return epoch_offset()
-
-        return normalize_datetime(dt).timestamp()
-
-
-def get_aware_datetime(dt=None, tz=UTC):
+def get_timestamp(dt=None):
     """
-    Returns an aware datetime object. If `dt` is not specified or `None`, the current time in UTC is assumed.
+    Returns the number of seconds since the Unix epoch.
+    If `dt` is not specified or `None`, the current time in UTC is assumed.
     """
     if dt is None:
-        return datetime.now(tz)
-    else:
-        return normalize_datetime(dt)
+        # NOTE: Although the epoch is platform dependent, it appears to be the same
```

Comment:
What do you mean "epoch is platform dependent"? You mean that some platforms are configured to ignore leap seconds or something?

I would expect `time.time()` to always return the unix epoch time.

Reply:
I was going by CPython's definition https://docs.python.org/3/library/time.html#epoch

Reply:
Are the docs inaccurate?

Reply:
Very interesting, I don't think I knew that. I would be very shocked if a system having a different epoch didn't break things horribly (though I guess some systems are configured to do the sane thing and ignore leap seconds — maybe that shows up differently in `time.time()`).

I'm curious to know what `datetime.timestamp()` would do on such a system. I guess it's also unclear what you would want to do on such a system, since I always think of a "timestamp" as defined relative to the Unix epoch, but possibly it could be defined relative to *any* epoch.

If you want to always use the system epoch, you could presumably change your definition of `EPOCH` to be `EPOCH = datetime(*time.gmtime(0)[0:6], tzinfo=UTC)`.

That said, that documentation seems weird. From what I can tell, the implementation of `time_gettime` on Windows uses a function that computes time from a totally different epoch (`1601-01-01`), and then adjusts it to `1970-01-01`. On non-Windows systems, it uses `clock_gettime` and `gettimeofday`, both of which are standardized in POSIX.1 as using the Unix epoch.

I'm wondering if we should change the `time.time` documentation to explicitly define it by the Unix epoch, rather than by the "system epoch".

Reply:
Thanks for explaining. I do think that the `time.time` documentation should be updated then, especially since it makes extra efforts to achieve POSIX compliance on Windows. I'll remove this comment.